pallets/utxo/src/lib.rs

```diff
@@ -903,11 +1078,46 @@ pub mod pallet {
 
             spend::<T>(&signer, &tx)
         }
+
+        /// unlock the stake, if user wants to stop validating and withdraw the locked utxos.
+        /// If used with `pallet-staking`, it uses the `BondingDuration`
+        /// to set the period/era on when to withdraw.
+        #[pallet::weight(T::WeightInfo::unlock_stake(1 as u32))]
+        pub fn unlock_stake(
+            origin: OriginFor<T>,
+            controller_pubkey: H256
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+            staking::unlock::<T>(&controller_pubkey)?;
```

Comment:
I'm not sure, but shouldn't the stash owner be the only authority allowed to unlock a stake? What's the point of delegating a stake if a controller can control even spending it?